src/stream/stream.rs

```diff
@@ -37,102 +38,122 @@ impl Stream {
             mavlink_camera,
         };
 
-        if let VideoSourceType::Redirect(_) = &video_and_stream_information.video_source {
-            // Do not add any Sink if it's a redirect Pipeline
-            return Ok(stream);
+        match &video_and_stream_information.video_source {
+            VideoSourceType::Redirect(_) => return Ok(stream), // Do not add any Sink if it's a redirect Pipeline
+            VideoSourceType::Gst(_) | VideoSourceType::Local(_) => (),
         }
 
-        // Add the desired Sinks to the Stream
-        let mut any_error: Result<()> = Ok(());
-        video_and_stream_information
-            .stream_information
-            .endpoints
-            .iter()
-            .for_each(|endpoint| {
-                let endpoint = endpoint.scheme();
-                let result = match endpoint {
-                    "udp" => {
-                        create_udp_sink(Manager::generate_uuid(), video_and_stream_information)
-                    }
-                    unsupported => Err(anyhow!("Unsupported Endpoint scheme: {unsupported}")),
-                };
-
-                if let Err(reason) = result.and_then(|sink| stream.pipeline.add_sink(sink)) {
-                    let error = anyhow!(
-                        "Failed to add Sink of type {endpoint} to the Pipeline. Reason: {reason}"
-                    );
-                    error!("{error:#?}");
-                    any_error = Err(error);
-                }
-            });
-        any_error?;
+        let endpoints = &video_and_stream_information.stream_information.endpoints;
+
+        if endpoints.iter().any(|endpoint| endpoint.scheme() == "udp") {
+            if let Err(reason) =
+                create_udp_sink(Manager::generate_uuid(), video_and_stream_information)
+                    .and_then(|sink| stream.pipeline.add_sink(sink))
+            {
+                return Err(anyhow!(
+                    "Failed to add Sink of type UDP to the Pipeline. Reason: {reason}"
+                ));
+            }
+        }
+
+        if endpoints.iter().any(|endpoint| endpoint.scheme() == "rtsp") {
+            if let Err(reason) =
+                create_rtsp_sink(Manager::generate_uuid(), video_and_stream_information)
+                    .and_then(|sink| stream.pipeline.add_sink(sink))
+            {
+                return Err(anyhow!(
+                    "Failed to add Sink of type RTSP to the Pipeline. Reason: {reason}"
+                ));
+            }
+        }
```

Comment:
The difference here is that we only want to add 1 UDP Sink (`multiudpsink`) even if we have several UDP endpoints. The same idea applies to RTSP, which, in the end only accepts 1 endpoint.